crates/latch_core/src/ecs/mod.rs

```diff
@@ -2,6 +2,70 @@
 //
 // Runtime-defined component system with TypeScript compatibility.
 // See docs/ecs-runtime-components.md for architectural details.
+//
+// # Architecture Overview
+//
+// The ECS is built around these core concepts:
+//
+// - **Entity**: A unique identifier (ID + generation + archetype + row)
+// - **Component**: POD data associated with entities (registered by ID, not Rust type)
+// - **Archetype**: A unique combination of component types
+// - **World**: Container for all entities and their components
+// - **Storage**: SoA (Structure-of-Arrays) storage per archetype
+//
+// # Usage Patterns
+//
+// ## Spawning Entities
+//
+// ```ignore
+// use latch_core::spawn;
+//
+// let entity = spawn!(world,
+//     Position { x: 1.0, y: 2.0 },
+//     Velocity { x: 0.5, y: 0.0 }
+// );
+// ```
+//
+// ## Single-Component Queries
+//
+// ```ignore
+// // Iterate over all entities with Position
+// for (entity, position) in world.query::<Position>() {
+//     println!("Entity {:?} at {:?}", entity, position);
+// }
+// ```
+//
+// ## Multi-Component Iteration
+//
+// ```ignore
+// use latch_core::{columns, columns_mut};
+//
+// // Process multiple components in parallel
+// world.par_for_each(&[Position::ID, Velocity::ID], |storage| {
```

Comment:
Prefer names like `for_each` over `par_for_each` and let parallelization just be a (hidden) implementation detail. From the perspective of our API users they just say "I want to iterate over all entities with components X, Y, and Z and update those components"